launcher/ui/dialogs/CreateShortcutDialog.cpp

```diff
@@ -0,0 +1,220 @@
+#include "minecraft/MinecraftInstance.h"
+#include "minecraft/ShortcutUtils.h"
+#include "minecraft/WorldList.h"
+#include "minecraft/auth/AccountList.h"
+
+CreateShortcutDialog::CreateShortcutDialog(InstancePtr instance, QWidget* parent)
+    : QDialog(parent), ui(new Ui::CreateShortcutDialog), m_instance(instance)
+{
+    ui->setupUi(this);
+
+    InstIconKey = instance->iconKey();
+    ui->iconButton->setIcon(APPLICATION->icons()->getIcon(InstIconKey));
+    ui->instNameTextBox->setPlaceholderText(instance->name());
+
+    auto mInst = std::dynamic_pointer_cast<MinecraftInstance>(instance);
+    m_QuickJoinSupported = mInst && mInst->traits().contains("feature:is_quick_play_singleplayer");
+    if (!m_QuickJoinSupported) {
+        ui->worldTarget->hide();
+        ui->worldSelectionBox->hide();
+        ui->serverTarget->setChecked(true);
+        ui->serverTarget->hide();
+        ui->serverLabel->show();
+    }
+
+    // Populate save targets
+    if (!DesktopServices::isFlatpak()) {
+        QString desktopDir = FS::getDesktopDir();
+        QString applicationDir = FS::getApplicationsDir();
+
+        if (!desktopDir.isEmpty())
+            ui->saveTargetSelectionBox->addItem("Desktop", QVariant::fromValue(SaveTarget::Desktop));
+
+        if (!applicationDir.isEmpty())
+            ui->saveTargetSelectionBox->addItem("Applications", QVariant::fromValue(SaveTarget::Applications));
+    }
+    ui->saveTargetSelectionBox->addItem("Other...", QVariant::fromValue(SaveTarget::Other));
+
+    // Populate worlds
+    if (m_QuickJoinSupported) {
+        auto worldList = mInst->worldList();
+        worldList->update();
+        for (const auto& world : worldList->allWorlds()) {
+            // Entry name: World Name [Game Mode] - Last Played: DateTime
+            QString entry_name = tr("%1 [%2] - Last Played: %3")
+                                     .arg(world.name(), world.gameType().toTranslatedString(), world.lastPlayed().toString(Qt::ISODate));
+            ui->worldSelectionBox->addItem(entry_name, world.name());
+        }
+    }
+
+    // Populate accounts
+    auto accounts = APPLICATION->accounts();
+    MinecraftAccountPtr defaultAccount = accounts->defaultAccount();
+    if (accounts->count() <= 0) {
+        ui->overrideAccountCheckbox->setEnabled(false);
+    } else
+        for (int i = 0; i < accounts->count(); i++) {
+            MinecraftAccountPtr account = accounts->at(i);
+            auto profileLabel = account->profileName();
+            if (account->isInUse())
+                profileLabel = tr("%1 (in use)").arg(profileLabel);
+            auto face = account->getFace();
+            QIcon icon = face.isNull() ? APPLICATION->getThemedIcon("noaccount") : face;
+            ui->accountSelectionBox->addItem(profileLabel, account->profileName());
+            ui->accountSelectionBox->setItemIcon(i, icon);
+            if (defaultAccount == account)
+                ui->accountSelectionBox->setCurrentIndex(i);
+        }
```

Comment:
Nit: if a block contains multiple lines it definitely should be surrounded by braces for readability
```suggestion
    } else {
        for (int i = 0; i < accounts->count(); i++) {
            MinecraftAccountPtr account = accounts->at(i);
            auto profileLabel = account->profileName();
            if (account->isInUse())
                profileLabel = tr("%1 (in use)").arg(profileLabel);
            auto face = account->getFace();
            QIcon icon = face.isNull() ? APPLICATION->getThemedIcon("noaccount") : face;
            ui->accountSelectionBox->addItem(profileLabel, account->profileName());
            ui->accountSelectionBox->setItemIcon(i, icon);
            if (defaultAccount == account)
                ui->accountSelectionBox->setCurrentIndex(i);
        }
    }
```